lib/synapse/config_generator/haproxy.rb

```diff
@@ -1136,28 +1142,36 @@ def update_backends_at(socket_file_path, watchers)
         end
       end
 
+      commands = []
+
       # actually enable the enabled backends, and disable the disabled ones
       cur_backends.each do |section, backends|
-        backends.each do |backend|
-          if enabled_backends.fetch(section, []).include? backend
-            command = "enable server #{section}/#{backend}\n"
+        backends.each do |backend, state|
+          if enabled_backends.fetch(section, Set.new).include? backend
+            next if state == 'UP'
```

Comment:
I think you want starts with UP since backends that are on their way up will be "UP 1/2" e.g.